pkg/csi/service/vanilla/controller.go

```diff
@@ -379,7 +380,25 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 			log.Errorf(errMsg)
 			return nil, status.Error(codes.NotFound, errMsg)
 		}
-		sharedDatastores, datastoreTopologyMap, err = c.nodeMgr.GetSharedDatastoresInTopology(ctx, topologyRequirement, c.manager.CnsConfig.Labels.Zone, c.manager.CnsConfig.Labels.Region)
+		vcenter, err := c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to get vCenter. Err: %v", err)
+			log.Errorf(errMsg)
+			return nil, status.Error(codes.NotFound, errMsg)
+		}
+		tagManager, err := cnsvsphere.GetTagManager(ctx, vcenter)
```

Comment:
Can we get the tag manager once and reuse for future API calls? Creating a tag manager in each create API call adds up to the latency.

Reply:
@SandeepPissay created an issue for this - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/667
@RaunakShah also provided the same comment here - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/650#discussion_r574854846

Reply:
In future, do we plan to make `TagManager` part of controller struct and reload it when it expires?

Reply:
Yes @shalini-b we make that change later as part of #667